src/main/java/co/cask/aws/s3/source/S3BatchSource.java

```diff
@@ -55,56 +55,70 @@ public S3BatchSource(S3BatchConfig config) {
     this.config = config;
   }
 
+  @Override
+  protected Map<String, String> getFileSystemProperties(BatchSourceContext context) {
+    String authenticationMethod = config.authenticationMethod;
+    Map<String, String> properties = new HashMap<>(config.getFilesystemProperties());
+    if (authenticationMethod != null && authenticationMethod.equalsIgnoreCase(ACCESS_CREDENTIALS)) {
+      if (config.path.startsWith("s3a://")) {
```

Comment:
So if it does not start with either and it looks like we don't validate it then it will fail at runtime

Reply:
nevermind we do validate it